libs/libcommon/src/libcommon/queue.py

```diff
@@ -238,6 +246,34 @@ def flat_info(self) -> FlatJobInfo:
         )
 
 
+DEFAULT_INCREASE_AMOUNT = 1
+DEFAULT_DECREASE_AMOUNT = -1
+
+
+class JobTotalMetricDocument(Document):
+    """Jobs total metric in mongoDB database, used to compute prometheus metrics.
+
+    Args:
+        queue (`str`): queue name
+        status (`str`): job status see libcommon.queue.Status
+        total (`int`): total of jobs
+        created_at (`datetime`): when the metric has been created.
+    """
+
+    id = ObjectIdField(db_field="_id", primary_key=True, default=ObjectId)
+    queue = StringField(required=True)
```

Comment:
maybe (queue, status) should be set as unique

Reply:
Done in https://github.com/huggingface/datasets-server/pull/1684/commits/721689f02a3fd58e117a8a4bf0033a5bd883d311